components/tasks/TaskTable.tsx

```diff
@@ -1,222 +0,0 @@
-import { components } from '@/lib/types/api';
-import { Button } from '@chakra-ui/button';
-import { Box, Flex, Heading } from '@chakra-ui/layout';
-import {
-    FormControl,
-    FormErrorMessage,
-    FormLabel,
-    Input,
-    Modal,
-    ModalCloseButton,
-    ModalContent,
-    ModalFooter,
-    ModalHeader,
-    ModalOverlay,
-    useDisclosure,
-} from '@chakra-ui/react';
-import { Table, Tbody, Td, Th, Thead, Tr } from '@chakra-ui/table';
-import React, { useState } from 'react';
-import { useSWRConfig } from 'swr';
-import { FormStatus } from '../projects/NewProject/reducer';
-import * as fetch from '@/lib/utils/fetch';
-import ErrorAlert from '../common/ErrorAlert';
-import useData from '@/lib/hooks/useData';
-import Loading from '../common/Loading';
-import checkDateOrder from '@/lib/utils/checkDateOrder';
-
-type TaskTableProps = {
-    project?: components['schemas']['ProjectDTO'];
-};
-
-type StateType = {
-    name: string;
-    description: string;
-    startDate: string;
-    endDate?: string;
-    formStatus: FormStatus;
-    errorMessage: string;
-};
-
-function TaskTable({ project }: TaskTableProps): JSX.Element {
-    const { isOpen, onOpen, onClose } = useDisclosure();
-    const [state, setState] = useState<StateType>({
-        name: '',
-        description: '',
-        startDate: new Date().toISOString().split('T')[0],
-        formStatus: FormStatus.IDLE,
-        errorMessage: '',
-    });
-
-    const {
-        data: tasks,
-        isError: taskError,
-        isLoading: tasksLoading,
-    } = useData<components['schemas']['TaskDTO'][]>('task', { projectId: `${project?.id}` });
```

Comment:
I don't think it's a good idea to make fetch requests in these kinds of components. I suggest that, as a rule of a thumb, anything in the `components` directory should not make fetch calls. Fetch calls should be made by components in `pages` directory and passed downwards as arguments in conditional render statements.